### PR TITLE
update full datelement

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Related Issue(s)
- #165

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by tba76 — APPROVED** [review]
🍺

File: src/Storage/Controllers/DataController.cs [hunk 1]
```diff
@@ -380,25 +379,31 @@ public async Task<ActionResult<DataElement>> OverwriteData(int instanceOwnerPart
...
+                updatedProperties.Add("/isRead", false);
             }
 
-            if (dataElement.Size > 0)
+            if (blobSize > 0)
             {
-                dataElement.FileScanResult = dataTypeDefinition.EnableFileScan ? FileScanResult.Pending : FileScanResult.NotApplicable;
+                FileScanResult scanResult = dataTypeDefinition.EnableFileScan ? FileScanResult.Pending : FileScanResult.NotApplicable;
+
+                updatedProperties.Add("/fileScanResult", scanResult);
```
Comment (acn-sbuad) [inline src/Storage/Controllers/DataController.cs:403]: toString on scanresult?

Comment (acn-sbuad) [reply src/Storage/Controllers/DataController.cs:403]: ToString not required. Verified in YT

File: src/Storage/Controllers/DataController.cs [hunk 2]
```diff
@@ -581,16 +596,16 @@ private async Task<ActionResult<DataElement>> InitiateDelayedDelete(Instance ins
         {
             DateTime deletedTime = DateTime.UtcNow;
 
-            dataElement.DeleteStatus = new()
+            var deleteStatus = new DeleteStatus()
```
Comment (SandGrainOne) [inline src/Storage/Controllers/DataController.cs:599]:
```suggestion
            DeleteStatus deleteStatus = new()
```

**Pull request merged**